### ISSUE TITLE
pairing filure when own_addr_type is BLE_ADDR_TYPE_PUBLIC in the BLE advertise parameters.

### ISSUE THREAD

**Issue by JoonDong2** [nkolban/esp32-snippets#346]
I'm testing gatt_security_server example in the ESP-IDF.

original is set own_addr_type to BLE_ADDR_TYPE_RANDOM when configuring the BLE advertise parameters. At this time, ESP32 advertise adv packet including random generated address instead of MAC address. Because of this, android phone cannot recognize bonded device. In other word, android phone recognizes it as a new device every time ESP32 is rebooted.

So I configured own_addr_type to BLE_ADDR_TYPE_PUBLIC, then ESP32 advertised adv packet including MAC address always. connection is done, but pairing procedure is failed when I did this. The message is below. I did not revise other configures.

ESP_GAP_BEL_SEC_REQ_EVT
ESP_GAP_BLE_AUTH_COMPL_EVT
remote BD_ADDR : 222248398ac0
address type = 0
pair status = fail
Bonded devices number : 0
Bonded devices list : 0
btm_sec_clr_temp_auth_service() - no dev CB
Device not found
ESP_GATTS_DISCONNECT_EVT

Is this a bug? I want them to pair naturally after ESP32 rebooted.

**Comment by chegewara** [comment]
Thanks for reporting, will investigate. If you can share code you are using it would be faster to investigate it.

**Comment by chegewara** [comment]
Ok, i see the problem. When you changed to use public address (mac address) you also need to change this line to not generate random address by commenting it out or set false:
```
//generate a resolvable random address
            esp_ble_gap_config_local_privacy(true);
```

**Comment by JoonDong2** [comment]
Thank you. It works on android nougat(7.1.1), but doesn't work on kitkat(4.4.2) when I configured to that.
On kitkat, pairing procedure is ok, but connection is destroyed immediately from second connection.
I'm not sure about other versions.

**Comment by chegewara** [comment]
Yes, im aware of that becuse im having the same issue with my android 5.0 device. Its reported to espressif. Thank you for your reports.

**Comment by JoonDong2** [comment]
On marshmallow(CM13.0), pairing procedure and re-connection is done, but connection state remains at "bonding..." in the nRF Connect app. Bonding information did not appear in the bluetooth setting. ESP32 save the pairing information.

**Comment by chegewara** [comment]
Should have been fixed now.
